lte/gateway/c/core/oai/tasks/amf/amf_authentication.cpp

```diff
@@ -761,28 +787,24 @@ int amf_proc_authentication_failure(amf_ue_ngap_id_t ue_id,
       if (MAX_SYNC_FAILURES <= auth_proc->retry_sync_failure) {
         rc = amf_auth_auth_rej(ue_id);
       } else {
-        struct tagbstring resync_param;
-        resync_param.data = (unsigned char*)calloc(1, RESYNC_PARAM_LENGTH);
-        if (resync_param.data == NULL) {
+        auth_proc->auts.data = (unsigned char*)calloc(1, RESYNC_PARAM_LENGTH);
+        if (auth_proc->auts.data == NULL) {
           OAILOG_ERROR(LOG_NAS_AMF,
                        "Sending authentication reject with cause "
                        "AMF_CAUSE_SYNCH_FAILURE\n");
           rc = amf_auth_auth_rej(ue_id);
           OAILOG_FUNC_RETURN(LOG_NAS_EMM, rc);
         }
 
-        memcpy(resync_param.data,
+        memcpy(auth_proc->auts.data,
                (amf_ctx->_vector[amf_ctx->_security.vector_index].rand),
                RAND_LENGTH_OCTETS);
 
-        memcpy((resync_param.data + RAND_LENGTH_OCTETS),
+        memcpy((auth_proc->auts.data + RAND_LENGTH_OCTETS),
                msg->auth_failure_ie.authentication_failure_info->data,
                AUTS_LENGTH);
 
-        start_authentication_information_procedure_synch(amf_ctx, auth_proc,
-                                                         &resync_param);
-        free_wrapper(reinterpret_cast<void**>(&resync_param.data));
-
+        start_authentication_information_procedure_synch(amf_ctx, auth_proc);
```

Comment:
Where is auth_proc->auts.data freed?

Reply:
https://github.com/magma/magma/blob/1b098d0b1a0c3179ca4145fac493eb1d42e0eafe/lte/gateway/c/core/oai/tasks/amf/nas_proc.cpp#L522